olp-cpp-sdk-dataservice-write/src/StreamLayerClientImpl.cpp

```diff
@@ -360,12 +372,11 @@ StreamLayerClientImpl::Flush() {
 
 olp::client::CancellationToken StreamLayerClientImpl::Flush(
     const StreamLayerClient::FlushCallback& callback) {
-  auto cancel_context = std::make_shared<CancellationContext>();
+  CancellationContext cancel_context;
 
-  // TODO: We should introduce an Executor abstraction to handle thread
-  // spwaning/pooling in a centralized/ configurable way
   auto self = shared_from_this();
-  auto flush_thread = std::thread([self, callback, cancel_context]() {
+
+  auto func = [=]() mutable {
```

Comment:
I think it is better to capture `cancel_context` explicitly, cause we had problem with 'capture everything' policy in other places, what do you think? 

It can be handy when debugging/reading the lambdas; it will be easier to track which actual object is used in lambda's body and which is not.

Reply:
As from my past experience when capturing explicit you should also add some warnings to compilation for unused captures else you could face some nasty memory leaks.
With default capture the compiler takes care that only what is used is captured. Usually there is no difference between them except you want to guard against using something that you should not. Might be that we had issues because we didn't use what should have been captured. But this is a design issue and we should not solve this by just capture individually.